### PR TITLE
setState text workaround solution added in README

### LINKED ISSUE #198: Document setState/text workaround
Issue #27 outlines a relatively common problem where changing the `text` doesn't update the widget.  There is a workaround solution (specify a `key` that changes based on the `text`), but it should be documented in the `README`.

I'd also like to understand _better_ what is going on, and _why_ this is necessary.

### PR DESCRIPTION
**Pull request by Mohitmadhav**
Fixes #198 

1. Wrote the `setState` `text` issue in the README file.
2. Added the solution (a `key` that changes based on the `text`) along with the video by Google Developers.

Please tell if anything else needs to be added  - Like flutter documentation, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -489,6 +489,8 @@ AnimatedTextKit(
 ),
 ```
 
+**Note:** You might come up with an issue that the `text` does not get updated with `setState` as shown [here](https://github.com/aagarwal1012/Animated-Text-Kit/issues/27).The solution to this, is a key that changes based on the text. For reference, watch [this](https://www.youtube.com/watch?v=kn0EOS-ZiIc) video.
```
Comment (aagarwal1012) [inline README.md:492]: Move this thing to the `Usage` section.

Comment (aagarwal1012) [reply README.md:492]: Add a space between -- `).The`

**Review by aagarwal1012 — COMMENTED** [review]
In terms of content, the documentation seems fine to me.

**Comment by Mohitmadhav** [comment]
I updated the README 2 minutes ago. Please check @aagarwal1012 @SirusCodes.

**Review by aagarwal1012 — APPROVED** [review]
lgtm

**Comment by aagarwal1012** [comment]
@all-contributors add @Mohitmadhav for documentation.

**Comment by SirusCodes** [comment]
Hey @aagarwal1012 have you seen my comment on?

**Comment by aagarwal1012** [comment]
@SirusCodes where is your comment?

**Comment by SirusCodes** [comment]
Actually I started a review

**Comment by aagarwal1012** [comment]
@SirusCodes why your comment doesn't appear in this PR?

**Comment by aagarwal1012** [comment]
Anyways, @Mohitmadhav can you create a new PR stating @SirusCodes' comments?

**Comment by SirusCodes** [comment]
Maybe starting a review won't appear on PR.🤔

**Comment by Mohitmadhav** [comment]
@SirusCodes so what more possible reasons could be there??
Even I am confused. Then only I'll be able to add an example with reference to that in the new PR.

**Comment by SirusCodes** [comment]
So, I tried to replicate the issue.. and observed a few things and found out the reason and I don't know exactly why keys fixes it. My 1st guess would be what that video says...

**What causes the issue:**
When user calls `setState` the tree changes and new text list is assigned but the `_index` field in the `AnimatedTextKit` is not updated according to new list size. That is if the user calls `setState` after the index is crossed it will throw `RangeError`.

**Example**
1. `text` has 3 strings
2. Animation completes till 2nd index
3. User calls `setState`
4. Now text length is 2 but index is 2 (on 3rd text)
5. `bool get _isLast => _index == widget.animatedTexts.length - 1;`  this condition fails and index is increased.
6. Which asks for index location 3 (4th text) which is not there in List hence causes the error.

**Possible fix**
`bool get _isLast => _index == widget.animatedTexts.length - 1;` => `bool get _isLast => _index >= widget.animatedTexts.length - 1;`
This might fix the issue but testing is required...

**Comment by SirusCodes** [comment]
@aagarwal1012 and @Mohitmadhav take a look at it

**Pull request merged**